<a href="https://colab.research.google.com/github/Kristian-Abad/Netflix-Movies-Project/blob/main/Netflix_Movies.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importing libraries

In [ ]:
import numpy as np
import pandas as pd

# Importing the data

The data comes from [this Kaggle dataset.](https://www.kaggle.com/datasets/amjaads/netflix-movies?resource=download)

In [ ]:
data = pd.read_csv('Netflix_movies.csv')

# Exploring our data

In [ ]:
data.head()

Unnamed: 0           movie_name  Duration  year  \
0           1    #FriendButMarried       102  2018   
1           4              #Selfie       125  2014   
2           5           #Selfie 69       119  2016   
3           7         #realityhigh        99  2017   
4           9  10 Days in Sun City        87  2017   

                                             genre          director  \
0    Dramas, International Movies, Romantic Movies     Rako Prijanto   
1           Comedies, Dramas, International Movies    Cristina Jacob   
2           Comedies, Dramas, International Movies    Cristina Jacob   
3                                         Comedies  Fernando Lebrija   
4  Comedies, International Movies, Romantic Movies         Adze Ugah   

                                              actors        country  rating  \
0  Adipati Dolken, Vanesha Prescilla, Rendi Jhon,...      Indonesia     7.0   
1  Flavia Hojda, Crina Semciuc, Olimpia Melinte, ...        Romania     6.1   
2  Maia Morgenstern, Olimpia Melinte, Crina Semci...        Romania     6.3   
3  Nesta Cooper, Kate Walsh, John Michael Higgins...  United States     5.2   
4  Ayo Makun, Adesua Etomi, Richard Mofe-Damijo, ...   South Africa     5.3   

    enter_in_netflix  
0       May 21, 2020  
1       June 1, 2019  
2       June 1, 2019  
3  September 8, 2017  
4   October 18, 2019

In [ ]:
data.loc[:,'Unnamed: 0']

0          1
1          4
2          5
3          7
4          9
        ... 
3318    3892
3319    3893
3320    3894
3321    3895
3322    3896
Name: Unnamed: 0, Length: 3323, dtype: int64

We're going to need to deal with this column later.

In [ ]:
data.shape

(3323, 10)

Okay so we're dealing with a dataframe with 3,323 records across 10 variables.

In [ ]:
data.isna().any()

Unnamed: 0          False
movie_name          False
Duration            False
year                False
genre               False
director            False
actors              False
country             False
rating              False
enter_in_netflix    False
dtype: bool

Looks like we are fortunate enough to not have any missing values.

In [ ]:
data = data.rename(columns={"Unnamed: 0" : "index"})
data.head()

index           movie_name  Duration  year  \
0      1    #FriendButMarried       102  2018   
1      4              #Selfie       125  2014   
2      5           #Selfie 69       119  2016   
3      7         #realityhigh        99  2017   
4      9  10 Days in Sun City        87  2017   

                                             genre          director  \
0    Dramas, International Movies, Romantic Movies     Rako Prijanto   
1           Comedies, Dramas, International Movies    Cristina Jacob   
2           Comedies, Dramas, International Movies    Cristina Jacob   
3                                         Comedies  Fernando Lebrija   
4  Comedies, International Movies, Romantic Movies         Adze Ugah   

                                              actors        country  rating  \
0  Adipati Dolken, Vanesha Prescilla, Rendi Jhon,...      Indonesia     7.0   
1  Flavia Hojda, Crina Semciuc, Olimpia Melinte, ...        Romania     6.1   
2  Maia Morgenstern, Olimpia Melinte, Crina Semci...        Romania     6.3   
3  Nesta Cooper, Kate Walsh, John Michael Higgins...  United States     5.2   
4  Ayo Makun, Adesua Etomi, Richard Mofe-Damijo, ...   South Africa     5.3   

    enter_in_netflix  
0       May 21, 2020  
1       June 1, 2019  
2       June 1, 2019  
3  September 8, 2017  
4   October 18, 2019

Since the now index column is non-monotonic let's just drop it.

In [ ]:
data = data.drop(columns='index')
# Verifying we have dropped the 'index' column
data.head()

movie_name  Duration  year  \
0    #FriendButMarried       102  2018   
1              #Selfie       125  2014   
2           #Selfie 69       119  2016   
3         #realityhigh        99  2017   
4  10 Days in Sun City        87  2017   

                                             genre          director  \
0    Dramas, International Movies, Romantic Movies     Rako Prijanto   
1           Comedies, Dramas, International Movies    Cristina Jacob   
2           Comedies, Dramas, International Movies    Cristina Jacob   
3                                         Comedies  Fernando Lebrija   
4  Comedies, International Movies, Romantic Movies         Adze Ugah   

                                              actors        country  rating  \
0  Adipati Dolken, Vanesha Prescilla, Rendi Jhon,...      Indonesia     7.0   
1  Flavia Hojda, Crina Semciuc, Olimpia Melinte, ...        Romania     6.1   
2  Maia Morgenstern, Olimpia Melinte, Crina Semci...        Romania     6.3   
3  Nesta Cooper, Kate Walsh, John Michael Higgins...  United States     5.2   
4  Ayo Makun, Adesua Etomi, Richard Mofe-Damijo, ...   South Africa     5.3   

    enter_in_netflix  
0       May 21, 2020  
1       June 1, 2019  
2       June 1, 2019  
3  September 8, 2017  
4   October 18, 2019

In [ ]:
data[data.genre == 'Comedies'].head()

movie_name  Duration  year     genre  \
3                   #realityhigh        99  2017  Comedies   
61          A Bad Moms Christmas       104  2017  Comedies   
77   A Futile and Stupid Gesture       102  2018  Comedies   
107             A Thousand Words        92  2012  Comedies   
149  Adventures in Public School        86  2018  Comedies   

                   director  \
3          Fernando Lebrija   
61   Jon Lucas, Scott Moore   
77               David Wain   
107           Brian Robbins   
149            Kyle Rideout   

                                                actors                country  \
3    Nesta Cooper, Kate Walsh, John Michael Higgins...          United States   
61   Mila Kunis, Kristen Bell, Kathryn Hahn, Susan ...   United States, China   
77   Will Forte, Domhnall Gleeson, Martin Mull, Joe...          United States   
107  Eddie Murphy, Kerry Washington, Cliff Curtis, ...          United States   
149  Daniel Doheny, Judy Greer, Siobhan Williams, R...  Canada, United States   

     rating   enter_in_netflix  
3       5.2  September 8, 2017  
61      5.6  February 14, 2020  
77      6.8   January 26, 2018  
107     5.9       July 1, 2020  
149     5.8    August 15, 2018

I want to compare the count of movies that are comedies (or contain comedy elements), horror (or contain horror elements), and possibly both as indicated by the 'genre' column by year entered in Netflix. Here's how I would initially like to tackle this problem:


*   Create a subset of the data containing only comedies.
*   Create a subset of the data containing only horror.
*   Create a subset of the data containing both.
*   Combine the three subsets
*   Group by year entered in netflix and genre. Then count, sum as a new column called count
*   Export for visualization in Tableau







Let's find all the movies that at least 'Comedies' in the genre column. The [reference](https://thecodingbot.com/check-if-a-column-contains-specific-string-in-a-pandas-dataframe/) for the function used.

In [ ]:
# Create a boolean filter of the records that contain 'Comedies' in the 'genre column'
comedies_filter = data.genre.str.contains('Comedies')

# Create a temporary subset of the data using the filter above
temp_subset = data[comedies_filter]

# Use the temporary subset to now filter out the "Horror" movies
data_comedies = temp_subset[~temp_subset.genre.str.contains('Horror')]

movie_name  Duration  year  \
1                               #Selfie       125  2014   
2                            #Selfie 69       119  2016   
3                          #realityhigh        99  2017   
4                   10 Days in Sun City        87  2017   
5                        10 jours en or        97  2012   
...                                 ...       ...   ...   
3308  Zipi & Zape y la Isla del Capitan       105  2016   
3313                     ¡Ay, mi madre!        81  2019   
3314                        Çarsi Pazar        97  2015   
3315        Çok Filim Hareketler Bunlar        99  2010   
3317                          الف مبروك       115  2009   

                                                genre           director  \
1              Comedies, Dramas, International Movies     Cristina Jacob   
2              Comedies, Dramas, International Movies     Cristina Jacob   
3                                            Comedies   Fernando Lebrija   
4     Comedies, International Movies, Romantic Movies          Adze Ugah   
5              Comedies, Dramas, International Movies  Nicolas Brossette   
...                                               ...                ...   
3308               Children & Family Movies, Comedies       Oskar Santos   
3313                   Comedies, International Movies        Frank Ariza   
3314                   Comedies, International Movies    Muharrem Gülmez   
3315                   Comedies, International Movies       Ozan Açıktan   
3317           Comedies, Dramas, International Movies  Ahmed Nader Galal   

                                                 actors        country  \
1     Flavia Hojda, Crina Semciuc, Olimpia Melinte, ...        Romania   
2     Maia Morgenstern, Olimpia Melinte, Crina Semci...        Romania   
3     Nesta Cooper, Kate Walsh, John Michael Higgins...  United States   
4     Ayo Makun, Adesua Etomi, Richard Mofe-Damijo, ...   South Africa   
5     Franck Dubosc, Claude Rich, Marie Kremer, Math...         France   
...                                                 ...            ...   
3308  Teo Planell, Toni Gómez, Elena Anaya, Iria Cas...          Spain   
3313  Estefanía de los Santos, Secun de la Rosa, Ter...          Spain   
3314  Erdem Yener, Ayhan Taş, Emin Olcay, Muharrem G...         Turkey   
3315  Ayça Erturan, Aydan Taş, Ayşegül Akdemir, Burc...         Turkey   
3317  Ahmed Helmy, Laila Ezz El Arab, Mahmoud El Fis...          Egypt   

      rating   enter_in_netflix  
1        6.1       June 1, 2019  
2        6.3       June 1, 2019  
3        5.2  September 8, 2017  
4        5.3   October 18, 2019  
5        5.8       July 1, 2017  
...      ...                ...  
3308     5.6       July 1, 2017  
3313     3.8      July 19, 2019  
3314     5.3     March 10, 2017  
3315     4.4     March 10, 2017  
3317     7.4     April 25, 2020  

[987 rows x 9 columns]

In [ ]:
# Verifying that we should get no entries that have 'Horror' in the genre
data_comedies[data_comedies.genre.str.contains('Horror')]

Empty DataFrame
Columns: [movie_name, Duration, year, genre, director, actors, country, rating, enter_in_netflix]
Index: []

Likewise we can do the same operations for our other subsets of the data:

In [ ]:
# Create a boolean filter of the records that contain 'Horror' in the 'genre column'
horror_filter = data.genre.str.contains('Horror')

# Create a temporary subset of the data using the filter above
temp_subset_2 = data[horror_filter]

# Use the temporary subset to now filter out the "Horror" movies
data_horror = temp_subset_2[~temp_subset_2.genre.str.contains('Comedies')]

In [ ]:
data_horror.head()

movie_name  Duration  year  \
9          122        95  2019   
10  13 Cameras        90  2015   
11     13 Sins        93  2014   
13  14 Cameras        89  2018   
17        1920       143  2008   

                                             genre  \
9              Horror Movies, International Movies   
10    Horror Movies, Independent Movies, Thrillers   
11                        Horror Movies, Thrillers   
13                        Horror Movies, Thrillers   
17  Horror Movies, International Movies, Thrillers   

                      director  \
9              Yasir Al Yasiri   
10              Victor Zarcoff   
11                Daniel Stamm   
13  Scott Hussion, Seth Fuller   
17                Vikram Bhatt   

                                               actors        country  rating  \
9   Amina Khalil, Ahmed Dawood, Tarek Lotfy, Ahmed...          Egypt     7.1   
10  PJ McCabe, Brianne Moncrief, Sarah Baldwin, Ji...  United States     5.1   
11  Mark Webber, Rutina Wesley, Devon Graye, Tom B...  United States     6.3   
13  Neville Archambault, Amber Midthunder, Brytnee...  United States     4.5   
17  Rajneesh Duggal, Adah Sharma, Indraneil Sengup...          India     6.4   

     enter_in_netflix  
9        June 1, 2020  
10    August 13, 2016  
11   January 13, 2019  
13   October 27, 2018  
17  December 15, 2017

In [ ]:
# Verifying that we should get no entries that have 'Horror' in the genre
data_horror[data_horror.genre.str.contains('Comedies')]

Empty DataFrame
Columns: [movie_name, Duration, year, genre, director, actors, country, rating, enter_in_netflix]
Index: []

In [ ]:
# Create a boolean filter of the records that contain both 'Comedies' and 'Horror' in the 'genre column'
ch_filter = data.genre.str.contains('Horror') & data.genre.str.contains('Comedies')

# Create a temporary subset of the data using the filter above
data_ch = data[ch_filter]
data_ch.head()

movie_name  Duration  year  \
81     A Haunted House        86  2013   
1047  Girls With Balls        77  2019   
1112   Hantu Kak Limah       110  2018   
1135     He Never Died        97  2015   
1173          Holidays       106  2016   

                                              genre  \
81                          Comedies, Horror Movies   
1047  Comedies, Horror Movies, International Movies   
1112  Comedies, Horror Movies, International Movies   
1135                        Comedies, Horror Movies   
1173                        Comedies, Horror Movies   

                                               director  \
81                                       Michael Tiddes   
1047                                     Olivier Afonso   
1112                                       Mamat Khalid   
1135                                     Jason Krawczyk   
1173  Anthony Scott Burns, Nicholas McCarthy, Adam E...   

                                                 actors          country  \
81    Marlon Wayans, Essence Atkins, Cedric the Ente...    United States   
1047  Victor Artus Solaro, Manon Azem, Louise Blachè...  France, Belgium   
1112  Awie, Arab Khalid, Sharwani Ns, Tj Isa, Delima...         Malaysia   
1135  Henry Rollins, Booboo Stewart, Kate Greenhouse...           Canada   
1173  Madeleine Coghlan, Savannah Kennick, Ruth Brad...    United States   

      rating   enter_in_netflix  
81       5.0  February 21, 2020  
1047     3.8      July 26, 2019  
1112     6.3   January 31, 2019  
1135     6.4     March 18, 2016  
1173     5.1      July 15, 2016

Taking a quick peak into the shape of our subsets:

In [ ]:
print('Comedies: ', data_comedies.shape,
      '\nHorror: ', data_horror.shape,'\n'
      'Comedies and Horror: ',data_ch.shape)

Comedies:  (987, 9) 
Horror:  (196, 9) 
Comedies and Horror:  (23, 9)
